cpp/src/community/k_truss_impl.cuh

```diff
@@ -656,282 +625,842 @@ k_truss(raft::handle_t const& handle,
-
-    auto prop_num_triangles = edge_triangle_count<vertex_t, edge_t, false>(handle, cur_graph_view);
-
-    std::tie(edgelist_srcs, edgelist_dsts, edgelist_wgts, num_triangles, std::ignore) =
-      decompress_to_edgelist(
-        handle,
-        cur_graph_view,
-        edge_weight_view,
-        // FIXME: Update 'decompress_edgelist' to support int32_t and int64_t values
-        std::make_optional(prop_num_triangles.view()),
-        std::optional<cugraph::edge_property_view_t<edge_t, int32_t const*>>{std::nullopt},
-        std::optional<raft::device_span<vertex_t const>>(std::nullopt));
-    auto transposed_edge_first =
-      thrust::make_zip_iterator(edgelist_dsts.begin(), edgelist_srcs.begin());
-
-    auto edge_first = thrust::make_zip_iterator(edgelist_srcs.begin(), edgelist_dsts.begin());
-
-    auto transposed_edge_triangle_count_pair_first =
-      thrust::make_zip_iterator(transposed_edge_first, (*num_triangles).begin());
-
-    thrust::sort_by_key(handle.get_thrust_policy(),
-                        transposed_edge_first,
-                        transposed_edge_first + edgelist_srcs.size(),
-                        (*num_triangles).begin());
+    
+    auto e_property_triangle_count = edge_triangle_count<vertex_t, edge_t, multi_gpu>(handle, cur_graph_view);
 
     cugraph::edge_property_t<decltype(cur_graph_view), bool> edge_mask(handle, cur_graph_view);
     cugraph::fill_edge_property(handle, cur_graph_view, true, edge_mask);
-    cur_graph_view.attach_edge_mask(edge_mask.view());
 
+    auto iteration = -1;
     while (true) {
-      // 'invalid_transposed_edge_triangle_count_first' marks the beginning of the edges to be
-      // removed 'invalid_transposed_edge_triangle_count_first' + edgelist_srcs.size() marks the end
-      // of the edges to be removed 'edge_triangle_count_pair_first' marks the begining of the valid
-      // edges.
-      auto invalid_transposed_edge_triangle_count_first =
-        thrust::stable_partition(handle.get_thrust_policy(),
-                                 transposed_edge_triangle_count_pair_first,
-                                 transposed_edge_triangle_count_pair_first + edgelist_srcs.size(),
-                                 [k] __device__(auto e) {
-                                   auto num_triangles = thrust::get<1>(e);
-                                   return num_triangles >= k - 2;
-                                 });
-      auto num_invalid_edges = static_cast<size_t>(
-        thrust::distance(invalid_transposed_edge_triangle_count_first,
-                         transposed_edge_triangle_count_pair_first + edgelist_srcs.size()));
-
-      if (num_invalid_edges == 0) { break; }
-
-      auto num_valid_edges = edgelist_srcs.size() - num_invalid_edges;
-
-      // case 1. For the (p, q), find intersection 'r'.
-
-      // nbr_intersection requires the edges to be sort by 'src'
-      // sort the invalid edges by src for nbr intersection
+      iteration += 1;
```

Comment:
Why not start with `size_t iteration = 0` and increase at the end? `iteration = -1` sounds weird (how can an iteration count be negative?)

Reply:
> Why not start with size_t iteration = 0 and increase at the end? iteration = -1 sounds weird (how can an iteration count be negative?)

Oh this was just for debugging purposes. I am not using `iteration` anywhere.